jpegutils.h

```diff
@@ -11,50 +11,8 @@
 #ifndef __JPEGUTILS_H__
 #define __JPEGUTILS_H__
 
- /*
-  * jpeg_data:       buffer with input / output jpeg
-  * len:             Length of jpeg buffer
-  * itype:           Y4M_ILACE_NONE: Not interlaced
-  *                  Y4M_ILACE_TOP_FIRST: Interlaced, top-field-first
-  *                  Y4M_ILACE_BOTTOM_FIRST: Interlaced, bottom-field-first
-  * ctype            Chroma format for decompression.
-  *                  Currently always 420 and hence ignored.
-  * raw0             buffer with input / output raw Y channel
-  * raw1             buffer with input / output raw U/Cb channel
-  * raw2             buffer with input / output raw V/Cr channel
-  * width            width of Y channel (width of U/V is width/2)
-  * height           height of Y channel (height of U/V is height/2)
-  */
+int jpgutl_decode_jpeg (unsigned char *jpeg_data, int len,
+                     unsigned int width_in, unsigned int height_in, unsigned char *img);
```

Comment:
Great!

I agree with you about unnecessary documentation; no point documenting an utterly obvious parameter, and renaming parameters to make them obvious sounds great too.

The jpeg_data parameter should probably be const btw, as I presume it's not modified in the function.


On documentation for functions: The projects I've worked on the rule has always been: if you're documenting something a caller of the function needs to know it goes in the .h; if you're documenting some implementation detail that is only relevant to someone who needs to modify the function then it goes in the .c.

I don't go looking for documentation in the .h either... my IDE (Xcode in this case, though Visual Studio etc have similar functionality) shows it automatically :-)